nimare/correct/correct.py

```diff
@@ -0,0 +1,138 @@
+    _correction_method = None
+
+    # Maps that must be available in the MetaResult instance
+    _required_maps = ('p',)
+
+    def __init__(self):
+        pass
+
+    @abstractproperty
+    def _name_suffix(self):
+        pass
+
+    def _validate_input(self, result):
+        if not isinstance(result, MetaResult):
+            raise ValueError("First argument to transform() must be an "
+                             "instance of class MetaResult, not {}."
+                             .format(type(result)))
+        for rm in self._required_maps:
+            if not result.maps.get(rm):
+                raise ValueError("{0} requires {1} maps to be present in the "
+                                 "MetaResult, but none were found."
+                                 .format(type(self), rm))
+
+    def _generate_secondary_maps(self, result, corr_maps):
+        # Generates corrected version of z and log-p maps if they exist
+        p = corr_maps['p']
+        if 'z' in result.maps:
+            corr_maps.maps['z'] = p_to_z(p) * np.sign(result.maps['z'])
+        if 'log_p' in result.maps:
+            corr_maps['log_p'] = -np.log10(p)
+        return corr_maps
+
+    def transform(self, result, **kwargs):
+        est = result.estimator
+        method = self._correction_method
+
+        # Make sure we return a copy of the MetaResult
+        result = result.copy()
+
+        # If a correction method with the same name exists in the current
+        # MetaEstimator, use it. Otherwise fall back on _transform.
+        if (method is not None and hasattr(est, method)):
+            corr_maps = getattr(est, method)(result, **kwargs)
+        else:
+            self._validate_input(result)
+            corr_maps = self._transform(result)
+
+        # Update corrected map names and add them to maps dict
+        corr_maps = {(k + self._name_suffix): v for k, v in corr_maps.items()}
+        result.maps.update(corr_maps)
+
+        return result
+
+    @abstractmethod
+    def _transform(self, result, **kwargs):
+        # Must return a dictionary of new maps to add to .maps, where keys are
+        # map names and values are the maps. Names must _not_ include
+        # the _name_suffix:, as that will be added in transform() (i.e.,
+        # return "p" not "p_corr-FDR_q-0.05_method-indep").
+        pass
+
+
+class FWECorrector(Corrector):
+    """
+    Perform family-wise error rate correction on a meta-analysis.
+
+    Parameters
+    ----------
+    voxel_thresh : `obj`:float
+        The FDR correction rate to use.
+    method : `obj`:str
+        The FWE correction to use. Either 'bonferroni' or 'permutation'.
+    """
+
+    _correction_method = '_fwe_correct'
+
+    def __init__(self, method='bonferroni'):
+        self.method = method
+
+    @property
+    def _name_suffix(self):
+        return '_corr-FWE_method-{}'.format(self.method)
+
+    def _transform(self, result, voxel_thresh=0.001):
```

Comment:
I think all arguments should be passed to `__init__` rather than `transform` unless there's some good reason not to. The `voxel_thresh` is IMO something that should be set at the `Corrector` level, not something that varies across `MetaResult` objects. (I assume this is just an oversight, since `voxel_thresh` isn't actually used anywhere at the moment.)

Reply:
Okay, that makes sense. I was having trouble accessing the arguments from init, but I can just put them in an attribute to access within transform, right? Like this:

```python

class FWECorrector(Corrector):
    """
    Perform family-wise error rate correction on a meta-analysis.

    Parameters
    ----------
    method : `obj`:str
        The FWE correction to use. Either 'bonferroni' or 'permutation'.
    **kwargs
        Keyword arguments to be used by the FWE correction implementation.
    """

    _correction_method = '_fwe_correct'

    def __init__(self, method='bonferroni', **kwargs):
        self.method = method
        self.parameters = kwargs
```